test/tests/dump.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:meilisearch/src/exception.dart';
+import 'package:test/test.dart';
+
+import '../utils/client.dart';
+
+Future<void> dumpTests() async {
+  setUpClient();
+
+  test('is created', () async {
+    final dump = await client.createDump();
+    expect(dump.keys, contains('uid'));
+    expect(dump.keys, contains('status'));
+    expect(dump['uid'], isNotEmpty);
+    expect(dump['status'], equals('in_progress'));
+  });
+
+  test('able to get dump status', () async {
+    final dump = await client.createDump();
+    final dumpStatus = await client.getDumpStatus(dump['uid']!);
+    expect(dumpStatus.keys, contains('uid'));
+    expect(dumpStatus.keys, contains('status'));
+    expect(dumpStatus['uid'], isNotEmpty);
+    expect(dumpStatus['status'], isNotEmpty);
+  });
+
+  test('does not exist error', () async {
+    expect(client.getDumpStatus('bad_uid'),
+        throwsA(isA<MeiliSearchApiException>()));
+  });
+}
```

Comment:
It is good to keep an empty line at the end of the files https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log